### PR TITLE
Update niscope system tests to use multi-instrument session

### LINKED ISSUE #1503: Test niscope API with multi-instrument session
### Description of issue

~Measurement library was added to niscope in #1485. We need to verify it works on multi-instrument session correctly.~

Verify niscope Python bindings work correctly with multi-instrument session.

### PR DESCRIPTION
**Pull request by sbethur**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

~- [ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~

- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

- Update niscope system tests to use multi-instrument session, wherever possible 
- Setup the test code such that when a new test is added, by default multi-instrument session gets used. Tests can opt to use `single_instrument_session`, if required.

### List issues fixed by this Pull Request below, if any.

* Fix #1503 

### What testing has been done?

Ran the system tests locally

### CODE REVIEW

File: src/niscope/system_tests/test_system_niscope.py [hunk 1]
```diff
@@ -8,6 +8,11 @@
 import tempfile
 
 
+instruments = ['FakeDevice1', 'FakeDevice2']
+# TODO(sbethur): Use `get_channel_names` when #1402 is fixed
```
Comment (marcoskirsch) [inline src/niscope/system_tests/test_system_niscope.py:12]: Nothing is stopping us from fixing this, right?

File: src/niscope/system_tests/test_system_niscope.py [hunk 2]
```diff
@@ -24,11 +29,17 @@
...
 
 @pytest.fixture(scope='function')
-def session():
+def single_instrument_session():
     with niscope.Session('FakeDevice', False, True, 'Simulate=1, DriverSetup=Model:5164; BoardType:PXIe') as simulated_session:
         yield simulated_session
 
 
+@pytest.fixture(scope='function')
+def session():
```
Comment (marcoskirsch) [inline src/niscope/system_tests/test_system_niscope.py:38]: consider calling this `multi_instrument_session`

File: src/niscope/system_tests/test_system_niscope.py (hunk 1 above)
Comment (sbethur) [reply src/niscope/system_tests/test_system_niscope.py:12]: That's right. We just need to prioritize it.

File: src/niscope/system_tests/test_system_niscope.py (hunk 2 above)
Comment (sbethur) [reply src/niscope/system_tests/test_system_niscope.py]: done

**Pull request merged**